rajawali/src/main/java/org/rajawali3d/materials/Material.java

```diff
@@ -1073,7 +1073,12 @@ public void setModelMatrix(Matrix4 modelMatrix) {
         mModelMatrix = modelMatrix;//.getFloatValues();
         mVertexShader.setModelMatrix(mModelMatrix);
 
-        mNormalMatrix.setAll(modelMatrix).setToNormalMatrix();
+        mNormalMatrix.setAll(modelMatrix);
+        try {
+            mNormalMatrix.setToNormalMatrix();
+        } catch (IllegalStateException exception) {
+            // modelMatrix is degenerate (zero scale)
```

Comment:
If it is done as a `Rajlog.d`, it will only show if `RajLog.setDebugEnabled(true)` is called, in which case the logs are going to be quite verbose anyway as all the materials will be printing their shader source, uniform handles, etc. The goal is just to give some indication to the developer, not the user, of what is happening, particularly when it happens for other reasons rather than the intentional use case you present. Additionally, it would not be bad to force the rest of the render process to be skipped.